store/src/java/com/zimbra/cs/mailbox/Mailbox.java

```diff
@@ -2895,7 +2900,7 @@ private PendingLocalModifications snapshotModifications(PendingLocalModification
                     }
                     snapshot.recordModified(folder, chg.why, (MailItem) chg.preModifyObj);
                 } else if (item instanceof Tag) {
-                    if (((Tag) item).isListed()) {
+                    if (((Tag) item).isListed() || item instanceof SmartFolder) {
```

Comment:
Is it worth having a `listedTagOrSmartFolder(item)` method that does this test as seems to be used a few times.

Reply:
Ah yeah that's a good idea.